cfgov/unprocessed/js/modules/Tree.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+/**
+ * Tree
+ * @class
+ *
+ * @classdesc A tree data structure.
+ * Tree's have one root node, and child nodes that branch.
+ * Like:
+ *
+ *        R
+ *       / \
+ *      A   B
+ *    / | \
+ *   C  D  E
+ *
+ * @returns {Tree} An instance.
+ */
+function Tree() {
+
+  var _root = null;
+  var _levelCache = {};
+
+  /**
+   * @param {Object} payload - Data to attach to the root node.
+   * @returns {Tree} An instance.
+   */
+  function init( payload ) {
+    _root = new TreeNode( this, payload );
+    _levelCache[0] = [ _root ];
+
+    return this;
+  }
+
+  /**
+   * @param {Object} payload - Data to attach to new node.
+   * @param {TreeNode} parent - Node on which to add a new child node.
+   * @returns {TreeNode} The newly instantiated and added node.
+   */
+  function add( payload, parent ) {
```

Comment:
I was going off of the advice [here](http://eslint.org/docs/rules/id-blacklist)
> Bad names can lead to hard to decipher code. Using generic names, such as data don’t infer much about the code and the values it receives.

It's specifically a something that gets carried around with a TreeNode, which seemed to be what payload described, but if that has another connotation, what do you think?